### PR TITLE
Rename CustomButton to TMButton

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
Remove existing TMButton
Rename CustomButton to TMButton
#114

### CODE REVIEW

File: DemoApp/DemoApp/Views/TMCardPage.xaml [hunk 1]
```diff
@@ -92,7 +93,7 @@
           <StackLayout BackgroundColor="Transparent"
                        HorizontalOptions="End"
                        Padding="0,10,0,10">
-            <local:TMButton Title="Button"></local:TMButton>
+            <Button:TMButton Title="Button" Size="Default"></Button:TMButton>
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TMCardPage.xaml:96]: Should be `Modus:TMButton`. All the controls should be exposed under the Trimble.Modus.Components namespace

File: Trimble.Modus.Components/Controls/TMModal/TMModal.cs [hunk 2]
```diff
@@ -149,8 +150,6 @@ private static void OnFullWidthButtonChanged(BindableObject bindable, object old
                     foreach (TMButton button in modal._buttonContainer.Children)
                     {
                         button.HorizontalOptions = LayoutOptions.FillAndExpand;
-                        button.frame.HorizontalOptions = LayoutOptions.FillAndExpand;
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/TMModal/TMModal.cs:152]: Are these intentional? @BAL4-R can you confirm?

**Review by rthanga1 — CHANGES_REQUESTED** [review]

**Review by rthanga1 — DISMISSED** [review]

File: Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs [hunk 3]
```diff
@@ -1,367 +1,367 @@
...
+                customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                customButton.buttonFrame.Stroke = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                customButton.buttonLabel.TextColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.White);
+                break;
+        }
+    }
+
+    private static void UpdateOutlineStyleColors(TMButton customButton)
+    {
+        customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent);
```
Comment (Sofiya-kumar) [inline Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs:275]: This method will return color only so casting not required

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs (hunk 3 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs]: Removed Casting

File: Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs [hunk 4]
```diff
@@ -1,367 +1,367 @@
...
+
+    private static void OnButtonColorChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMButton customButton && !customButton.IsFloatingButton)
+        {
+            CheckButtonStyle(customButton);
+        }
+    }
+
+    private static void CheckButtonStyle(TMButton customButton)
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs]: Renamed

File: Trimble.Modus.Components/Controls/TMModal/TMModal.cs (hunk 2 above)
Comment (ElishaSamPeterPrabhu) [reply Trimble.Modus.Components/Controls/TMModal/TMModal.cs:152]: Yes the FillAndExpand sets all the buttons in modal to be fullwidth

File: DemoApp/DemoApp/Views/TMCardPage.xaml [hunk 5]
```diff
@@ -92,7 +92,7 @@
           <StackLayout BackgroundColor="Transparent"
                        HorizontalOptions="End"
                        Padding="0,10,0,10">
-            <local:TMButton Title="Button"></local:TMButton>
+            <local:TMButton Title="Button" Size="Default"></local:TMButton>
```
Comment (Sofiya-kumar) [inline DemoApp/DemoApp/Views/TMCardPage.xaml:95]: Why default added here?

**Review by Sofiya-kumar — CHANGES_REQUESTED** [review]

File: DemoApp/DemoApp/Views/TMCardPage.xaml (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMCardPage.xaml:95]: If we don't set size "default" ,size is  not set by default .

Comment (Sofiya-kumar) [reply DemoApp/DemoApp/Views/TMCardPage.xaml:95]: I think we have to set it in TMButton.xaml.cs size value as default

**Review by rthanga1 — DISMISSED** [review]

File: DemoApp/DemoApp/Views/TMCardPage.xaml (hunk 5 above)
Comment (ElishaSamPeterPrabhu) [reply DemoApp/DemoApp/Views/TMCardPage.xaml]: Done

File: Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs [hunk 6]
```diff
@@ -36,7 +35,7 @@ public partial class TMButton : ContentView
         BindableProperty.Create(nameof(CommandParameter), typeof(object), typeof(TMButton), null);
 
     public static readonly BindableProperty SizeProperty =
-        BindableProperty.Create(nameof(Size), typeof(Enums.Size), typeof(TMButton), propertyChanged: OnSizeChanged);
+        BindableProperty.Create(nameof(Size), typeof(Enums.Size), typeof(TMButton), defaultValue: Enums.Size.Empty, propertyChanged: OnSizeChanged);
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs:38]: Can't you give `Enums.Size.Default` directly here?

**Review by rthanga1 — DISMISSED** [review]

File: Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs (hunk 6 above)
Comment (rthanga1) [reply Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs:38]: Ok

**Review by Sofiya-kumar — APPROVED** [review]

**Review by Sofiya-kumar — APPROVED** [review]

**Review by Sofiya-kumar — APPROVED** [review]

**Pull request merged**